src/libpmempool/pool.c

```diff
@@ -69,6 +66,15 @@
 #include "util_pmem.h"
 #include "mmap.h"
 
+#ifdef __FreeBSD__
+/*
+ * This include is ordered last so that our local copy of queue.h
+ * will be found instead of the FreeBSD system copy.
```

Comment:
This is an unacceptable hack. It relies on the name of the macro in FreeBSD's header guard.
What actually needs to happen is that we have to rename macros exported by our queue.h (including the header guard).

I think we should merge other changes in this PR as is and you should propose a new PR with just queue.h related changes.